inception-project/inception-plugin-server
# #16, 17, 25, 38, 39, 41, 42 - plugin database

### This is the last pull request that only contains code that was written during the BP.

**What's in the PR**
* UI for uploading, managing and downloading plugins
* a database that stores plugins and their metadata

**How to test manually**
* Log on, click "My Plugins", upload a file. If the file is a *.jar or *.zip file and the form contains correct info, the plugin is stored in the database. If the file does not have a valid ZIP header or if any entries in the form have the wrong format, the new plugin or plugin version is not created.
* After uploading, click "Browse Plugins" on the Dashboard. You should see a list of plugins with an entry for the plugin you jst created. Click the link it to go to the download page. The download button works, but the "download with dependencies" button creates an empty *.zip file.

**Automatic testing**

**Documentation**
* [ ] PR updates documentation (unless JavaDoc comments count)
